pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -543,23 +543,32 @@ private boolean processStreamEvents(MessageBatch messagesAndOffsets, long idlePi
       // Decode message
       StreamDataDecoderResult decodedRow = _streamDataDecoder.decode(messagesAndOffsets.getStreamMessage(index));
       RowMetadata msgMetadata = messagesAndOffsets.getStreamMessage(index).getMetadata();
+      GenericRow decoderResult;
       if (decodedRow.getException() != null) {
```

Comment:
qq: Currently, stream data decoder decodes key, value, headers and metadata (See `StreamDataDecoderImpl`). Should Pinot treat failure to decode each of these parts similarly? 
Previously, Pinot's decode failure would only pertain to the value decoding errors. If the user is not interested in the header/metadata/key fields, I wonder if there will be a case where this partially decoded result is still needed and it should not be considered as a failure. What are your thoughts on that?

I think it might be cleaner to wrap this `continueOnError` logic within `StreamDataDecoderImpl` ? Or extend that class and handle?

Reply:
Makes sense. We can treat metadata vs data seperately. Are there any cases where metadata errors are critical?

Reply:
I think header issues might be more critical as metadata is usually populated by the stream plugin itself and should be standardized for each.
I took another look at the code and realized that we just add the header object into the GenericRow. There is no actual decoding happening here. Not sure how this will be transformed in the transform pipeline. 
```
        RowMetadata metadata = message.getMetadata();
        if (metadata != null) {
          metadata.getHeaders().getFieldToValueMap()
              .forEach((key, value) -> row.putValue(HEADER_KEY_PREFIX + key, value));

          metadata.getRecordMetadata()
                  .forEach((key, value) -> row.putValue(METADATA_KEY_PREFIX + key, value));
        }
```

Reply:
In transform pipeline, the errors are already handled gracefully so that is not a concern at all.